src/ledger/GeneralizedLedgerEntry.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#pragma once
+
+#include "xdr/Stellar-ledger-entries.h"
+#include "xdr/Stellar-ledger.h"
+
+namespace stellar
+{
+
+enum class GeneralizedLedgerEntryType
+{
+    LEDGER_ENTRY
+};
+
+class GeneralizedLedgerKey
+{
+  private:
+    GeneralizedLedgerEntryType mType;
+    union {
+        LedgerKey mLedgerKey;
+    };
+
+    void assign(GeneralizedLedgerKey const& glk);
+    void construct();
+    void destruct();
+
+    void checkDiscriminant(GeneralizedLedgerEntryType expected) const;
+
+  public:
+    GeneralizedLedgerKey(LedgerKey const& lk);
+    GeneralizedLedgerKey(GeneralizedLedgerKey const& glk);
```

Comment:
no "move" constructor?

Reply:
It'll be in the next update. I've been plugging all the holes as I go. I'm also adding a default constructor and a constructor that takes `GeneralizedLedgerEntryType`.